pkg/controller/controller_binding.go

```diff
@@ -349,38 +352,70 @@ func isPlanBindable(serviceClass *v1alpha1.ServiceClass, plan *v1alpha1.ServiceP
 
 func (c *controller) injectBinding(binding *v1alpha1.Binding, credentials map[string]interface{}) error {
 	glog.V(5).Infof("Creating Secret %v/%v", binding.Namespace, binding.Spec.SecretName)
-	secret := &v1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      binding.Spec.SecretName,
-			Namespace: binding.Namespace,
-		},
-		Data: make(map[string][]byte),
-	}
 
+	secretData := make(map[string][]byte)
 	for k, v := range credentials {
 		var err error
-		secret.Data[k], err = serialize(v)
+		secretData[k], err = serialize(v)
 		if err != nil {
 			return fmt.Errorf("Unable to serialize credential value %q: %v; %s",
 				k, v, err)
 		}
 	}
 
-	found := false
+	secretClient := c.kubeClient.Core().Secrets(binding.Namespace)
+	// Retry loop for upserting the Secret
+	for {
```

Comment:
> Use one of the util.Wait methods here instead of making an infinite loop.

agreed, will rewrite.

> For now, I think I'd want to just do a single retry. Once we have a terminal condition for binding, we can set that terminal condition.

Why single retry? Isn't timeout in util.Wait enough?

In case of terminal condition there is already escape from the loop straight away, see [controller_binding.go#L381](https://github.com/atlassian/service-catalog/blob/c139e1810b63e3124fb6617719f68a270130fe3c/pkg/controller/controller_binding.go#L381) and [controller_binding.go#L409](https://github.com/atlassian/service-catalog/blob/c139e1810b63e3124fb6617719f68a270130fe3c/pkg/controller/controller_binding.go#L409)